pyroomacoustics/directivities.py

```diff
@@ -0,0 +1,183 @@
+import numpy as np
+from pyroomacoustics.utilities import Options
+
+
+class DirectivityPattern(Options):
+    FIGURE_EIGHT = "figure_eight"
+    HYPERCARDIOID = "hypercardioid"
+    CARDIOID = "cardioid"
+    SUBCARDIOID = "subcardioid"
+    OMNI = "omni"
+
+
+PATTERN_TO_CONSTANT = {
+    DirectivityPattern.FIGURE_EIGHT: 0,
+    DirectivityPattern.HYPERCARDIOID: 0.25,
+    DirectivityPattern.CARDIOID: 0.5,
+    DirectivityPattern.SUBCARDIOID: 0.75,
+    DirectivityPattern.OMNI: 1.0
+}
+
+
+class DirectionVector(object):
```

Comment:
For this one, I am a little bit ambivalent. It makes things quite clear and let us run some checks, but at the same time, it makes the machinery more heavy, compared to just using numpy arrays.

I think there are different solutions:
1. Just use a function to create unit vectors from spherical coordinates. I think we already have them in the doa sub-package, so we could just pull them somewhere more obvious.
2. Make `DirectionVector` a [sub-class of ndarray](https://numpy.org/doc/stable/user/basics.subclassing.html). This is quite neat. We can make sure the vector is normalized in the constructor, and then, we can still use this class as input to any numpy method. We can also add methods to convert back to azimuth/colatitude. This could also allow us to deal with the vectorized case by for example allowing extra dimensions.
3. Use plain numpy arrays...

Reply:
I agree, it makes it clunky and harder to vectorize. I've kept this `DirectionVector` for when creating a `CardioidFamily` object so that `azimuth` and `colatitude` (and the corresponding unit vector) are bundled into a single object.

Reply:
Sounds good!